### PR TITLE
Implement isStandbyDelegate function in DPoS - Closes #4936

### LINKED ISSUE #4936: Update DPoS library to provide information of an address is standby delegate or not
### Description
Update DPoS library to expose a function to check if a delegate is a standby delegate or not at particular height.
`isStandby(address:string, height: number, stateStore: StateStore) => boolean`.

forgersList for each round contains `standby: [address]`, so it can be calculated from there

### Motivation
In the BFT, it should not include the standby delegate for the prevote and pre commit calculation.
### Acceptance Criteria
- Test case when the block is 3 rounds old, which is maximum according to the BFT rule.
- Test case when checking the latest block and forged by standby delegate
- Test case when checking the latest block and forged by active delegate

### Related issues
#4915

### PR DESCRIPTION
**Pull request by ishantiw**
### What was the problem?

This PR resolves #4936 

### How was it solved?

- [x] Implemented `isStandByDelegate` in `lisk-dpos`

```typescript
public async isStandByDelegate(
		address: string,
		height: number,
		stateStore: StateStore,
	): Promise<boolean> 
```

- [x] Test case when the block is 3 rounds old, which is maximum according to the BFT rule.
- [x] Test case when checking the latest block and forged by standby delegate
- [x] Test case when checking the latest block and forged by active delegate

### How was it tested?

Run ` npm run test is_standby_delegate` under `elements/lisk-dpos`

### CODE REVIEW

File: elements/lisk-dpos/src/dpos.ts [hunk 1]
```diff
@@ -148,6 +149,28 @@ export class Dpos {
...
+		height: number,
+		stateStore: StateStore,
+	): Promise<boolean> {
+		const forgersList = await getForgersList(stateStore);
+		const relevantRound = this.rounds.calcRound(height);
+		const foundForgerList = forgersList.find(
+			roundRecord => roundRecord.round === relevantRound,
+		);
+
+		if (!foundForgerList) {
```
Comment (shuse2) [inline elements/lisk-dpos/src/dpos.ts:163]: I think if no relevant forger list, it should an error (it is something wrong with the state

File: elements/lisk-dpos/src/dpos.ts [hunk 2]
```diff
@@ -148,6 +149,28 @@ export class Dpos {
...
+			return false;
+		}
+
+		const isStandby = foundForgerList.standby.find(
+			standByDelegate => getAddressFromPublicKey(standByDelegate) === address,
```
Comment (shuse2) [inline elements/lisk-dpos/src/dpos.ts:168]: this is saved as address, so you can compare directly

File: elements/lisk-dpos/test/unit/is_standby_delegate.spec.ts [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+			const isStandby = await dpos.isStandByDelegate(
+				standByAddress,
+				height,
+				createStateStore(lists),
+			);
+			// Assert
+			expect(isStandby).toEqual(false);
+		});
+
+		it('should return false if a delegate is a normal delegate ', async () => {
```
Comment (shuse2) [inline elements/lisk-dpos/test/unit/is_standby_delegate.spec.ts:98]:
```suggestion
		it('should return false if a delegate is an active delegate ', async () => {
```

File: elements/lisk-dpos/src/dpos.ts (hunk 2 above)
Comment (ishantiw) [reply elements/lisk-dpos/src/dpos.ts:168]: I was confused by the `generateDelegateLists` utils, that adds a publicKey in the forgers list

Comment (ishantiw) [reply elements/lisk-dpos/src/dpos.ts]: I've updated `generateDelegateLists` function as well in `utils`

File: elements/lisk-dpos/test/unit/get_min_height_active.spec.ts [hunk 4]
```diff
@@ -73,7 +73,7 @@ describe('dpos.getMinActiveHeight()', () => {
 			height = 15 * ACTIVE_DELEGATES;
 
 			const lists = generateDelegateLists({
-				publicKey,
+				address: publicKey,
```
Comment (shuse2) [inline elements/lisk-dpos/test/unit/get_min_height_active.spec.ts:76]: maybe better to change the variable name to `address`?

Comment (ManuGowda) [inline elements/lisk-dpos/test/unit/get_min_height_active.spec.ts:76]: publicKey value is passed as address?

File: elements/lisk-dpos/test/utils/delegates.ts [hunk 5]
```diff
@@ -16,39 +16,56 @@ import { deepFreeze } from './deep_freeze';
...
 				delegates: [publicKey, ...list.delegates].slice(0, 3),
+				standby: [],
+			};
+		}
+		return list;
+	});
+};
+
+export const generateDelegateListsWithStandby = (
+	{ address: publicKey, activeRounds }: ActiveDelegateList,
```
Comment (shuse2) [inline elements/lisk-dpos/test/utils/delegates.ts:60]:
```suggestion
	{ address, activeRounds }: ActiveDelegateList,
```

File: elements/lisk-dpos/test/utils/delegates.ts [hunk 6]
```diff
@@ -16,39 +16,56 @@ import { deepFreeze } from './deep_freeze';
...
+	lists = delegateLists,
+): ForgersList => {
+	return lists.map((list: ForgerList) => {
+		if (activeRounds.includes(list.round)) {
+			return {
+				round: list.round,
+				delegates: [...list.delegates].slice(0, 3),
+				standby: [publicKey],
```
Comment (shuse2) [inline elements/lisk-dpos/test/utils/delegates.ts:68]:
```suggestion
				standby: [address],
```

**Comment by ishantiw** [comment]
@ManuGowda @shuse2 Addressed.

**Pull request merged**